server/src/internalClusterTest/java/org/elasticsearch/search/fetch/subphase/highlight/HighlighterSearchIT.java

```diff
@@ -1161,50 +1146,35 @@ private void checkMatchedFieldsCase(boolean requireFieldMatch, String type) thro
             response -> assertHighlight(response, 0, "foo", 0, equalTo(expectedHighlight0))
         );
 
-        // Now make sure boosted fields don't blow up when matched fields is both the subfield and stored field.
         fooField = new Field("foo").numOfFragments(1)
             .order("score")
             .fragmentSize(25)
             .highlighterType(type)
             .requireFieldMatch(requireFieldMatch);
         fooField.matchedFields("foo", "foo.plain");
         req = prepareSearch("test").highlighter(new HighlightBuilder().field(fooField));
-        assertResponse(
-            req.setQuery(queryStringQuery("foo.plain:running^5 scissors").field("foo")),
-            response -> assertHighlight(response, 0, "foo", 0, equalTo("<em>running</em> with <em>scissors</em>"))
-        );
 
-        // Now just all matches are against the matched field. This still returns highlighting.
-        assertResponse(
+        assertResponses(
+            response -> assertHighlight(response, 0, "foo", 0, equalTo("<em>running</em> with <em>scissors</em>")),
+            // Now make sure boosted fields don't blow up when matched fields is both the subfield and stored field.
+            req.setQuery(queryStringQuery("foo.plain:running^5 scissors").field("foo")),
```

Comment:
Same as above; are we reusing the same `Request` instance in these tests?

Reply:
Done!